src/cluster.py

```diff
@@ -511,6 +518,61 @@ def start_patroni(self) -> bool:
             logger.exception(error_message, exc_info=e)
             return False
 
+    def is_pitr_failed(self) -> bool:
+        """Check if Patroni service is down and there is fatal error during cluster bootstrap process in its logs.
+
+        In restore action, this means that database service failed to reach point-in-time-recovery target or has been
+        supplied with bad PITR parameter. Executes only on current unit.
+        """
+        patroni_logs = self._patroni_logs()
+        if "patroni.exceptions.PatroniFatalException: Failed to bootstrap cluster" in patroni_logs:
+            postgresql_logs = self._last_postgresql_logs()
+            if (
+                "FATAL:  recovery ended before configured recovery target was reached"
+                in postgresql_logs
+            ):
```

Comment:
Does this make any difference, as this method is already returning `True` at the same level as this if block?

Reply:
Fixed